tests/test_router_bgp.rb

```diff
@@ -223,106 +232,95 @@ def test_routerbgp_destroy
-             'bgp bestpath_med_missing_as_worst should be enabled')
-      if platform == :nexus
-        # TODO: only applies to :nexus
-        bgp.bestpath_med_non_deterministic = true
-        assert(bgp.bestpath_med_non_deterministic, "vrf #{vrf}: "\
-               'bgp bestpath_med_non_deterministic should be enabled')
-      end
-      bgp.bestpath_always_compare_med = false
-      refute(bgp.bestpath_always_compare_med, "vrf #{vrf}: "\
-             'bgp bestpath_always_compare_med should be disabled')
-      bgp.bestpath_aspath_multipath_relax = false
-      refute(bgp.bestpath_aspath_multipath_relax, "vrf #{vrf}: "\
-             'bgp bestpath_aspath_multipath_relax should be disabled')
-      bgp.bestpath_compare_routerid = false
-      refute(bgp.bestpath_compare_routerid, "vrf #{vrf}: "\
-             'bgp bestpath_compare_routerid should be disabled')
-      bgp.bestpath_cost_community_ignore = false
-      refute(bgp.bestpath_cost_community_ignore, "vrf #{vrf}: "\
-             'bgp bestpath_cost_community_ignore should be disabled')
-      bgp.bestpath_med_confed = false
-      refute(bgp.bestpath_med_confed, "vrf #{vrf}: "\
-             'bgp bestpath_med_confed should be disabled')
-      bgp.bestpath_med_missing_as_worst = false
-      refute(bgp.bestpath_med_missing_as_worst, "vrf #{vrf}: "\
-             'bgp bestpath_med_missing_as_worst should be disabled')
-      if platform == :nexus
-        # TODO: Only applies to :nexus
-        bgp.bestpath_med_non_deterministic = false
-        refute(bgp.bestpath_med_non_deterministic, "vrf #{vrf}: "\
-             'bgp bestpath_med_non_deterministic should be disabled')
-      end
-      bgp.destroy
+  def test_bestpath_default
+    bestpath(setup_default)
+  end
+
+  def test_bestpath_vrf
+    bestpath(setup_vrf)
+  end
+
+  def bestpath(bgp)
+    bgp.bestpath_always_compare_med = true
+    assert(bgp.bestpath_always_compare_med, "vrf #{@vrf}: "\
+           'bgp bestpath_always_compare_med should be enabled')
+    bgp.bestpath_aspath_multipath_relax = true
+    assert(bgp.bestpath_aspath_multipath_relax, "vrf #{@vrf}: "\
+           'bgp bestpath_aspath_multipath_relax should be enabled')
+    bgp.bestpath_compare_routerid = true
+    assert(bgp.bestpath_compare_routerid, "vrf #{@vrf}: "\
+           'bgp bestpath_compare_routerid should be enabled')
+    bgp.bestpath_cost_community_ignore = true
+    assert(bgp.bestpath_cost_community_ignore, "vrf #{@vrf}: "\
+           'bgp bestpath_cost_community_ignore should be enabled')
+    bgp.bestpath_med_confed = true
+    unless platform == :ios_xr && !@vrf == 'default'
+      assert(bgp.bestpath_med_confed, "vrf #{@vrf}: "\
+             'bgp bestpath_med_confed should be enabled')
+    end
+    bgp.bestpath_med_missing_as_worst = true
+    assert(bgp.bestpath_med_missing_as_worst, "vrf #{@vrf}: "\
+           'bgp bestpath_med_missing_as_worst should be enabled')
+    if platform == :nexus
```

Comment:
If these properties are unsupported on XR, I'd like to see that tested as well, i.e.:

```ruby
if platform == :nexus
  ...
else
  assert_raises(Cisco::UnsupportedError) { bgp.bestpath_med_non_deterministic = true }
```